styleguide/template.md

```diff
@@ -222,44 +222,34 @@ You can use a [Markdown table generator tool](http://www.tablesgenerator.com/mar
 The best way to include code is to include snippets from a working sample. Create your
 sample following the instructions in the [contributing guide](../CONTRIBUTING.md#contributing-to-samples).
 
-You can include the code using include syntax:
-
-```
-[!code-csharp[<title>](<pathToFile>#<RegionName)]
+You can include the code using the following syntax:
+```markdown
+[!code-<language>[<name>](<pathToFile><queryoption><queryoptionvalue>)]
 ```
 
-The example above shows C# syntax, but other languages are supported.
-Use `code-fsharp` for F# samples; use `code-vbnet` for Visual Basic samples.
-Other languages that are supported are:
-* C++: `code-cpp`
-* HTML: `code-html`
-* JavaScript: `code-javascript`
-* Powershell: `code-ps`
-* SQL: `code-sql`
-* XML: `code-xml`
-
-
+* `-<language>` (*optional* but *recommended*)
+  * Language of the code snippet being referenced. For a list of supported values, see [Supported languages](#supported-languages).
 
-The text you place for `<title>` shows up as a rollover on the text. The `<pathToFile>`
-is the path to the source file. The `<RegionName>` should be a region in your source
-code that should be included. Use the `#region` and `#endregion` preprocessor syntax
-to specify the region of code to include.
+* `<name>` (*optional*) 
+  * Name for the code snippet. It doesn’t have any impact on the output HTML, but you can use it to improve the readability of your Markdown source. 
 
-For cases where regions don't work, you can specify the start and end of a snippet
-using an XML element name in a single line comment. For example, you could write this in C#:
+* `<pathToFile>` (*mandatory*) 
+  * Relative path in the file system that indicates the code snippet file to reference.
+  
+* `<queryoption>` and `<queryoptionvalue>` (*optional*)
+ * Used together to specify how the code should be retrieved from the file:
+  * `#`:  `#L{startlinenumber}-L{endlinenumber}` (line range) *or* `#{tagname}` (tag name)
```

Comment:
Although it's mentioned later, perhaps mention here also that line numbers are discouraged?

Reply:
I'm moving that sentence to here then.